charts/kubechecks-rbac/templates/role.yaml

```diff
@@ -0,0 +1,11 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ .Values.roleName | default "kubechecks-remote-role" }}
```

Comment:
IMO, kubechecks staging should be deployed to tooling EKS then. This increases the complexity of kubechecks because now, we have to worry about external cluster dependencies, authentication and refreshing credentials 🤔 . @djeebus  what do you think?

Reply:
In the future, I think we do want to support the idea of hosting kubechecks outside of the cluster that argocd is running in. I'm not opposed to this change, mostly because this is opt-in behavior, and if you do opt-in, it's going to be complicated one way or another.

Reply:
It's definitely a nice to have. I'm just looking at the complexity it'll require to setup. It's a gate that once opened will flood in lots of requirements for different cluster setups, eks with OIDC, eks with rancher setup etc. If we can avoid inter-cluster auth, I think we should 😄. I'm just being too careful to make sure we are not the ones to open pandoras box ourself. Maybe allow for folks to request it?

Reply:
The /pkg/kubernetes package currently supports both EKS and in-cluster setups. To call the backend API for EKS, the caller must have an assigned IAM role. This is a straightforward use case. (sysadmin can assign role using kube2iam/IRSA, or even use env var to get the IAM role)

Regarding OIDC or Rancher, they do not impact the current functionality described above.

In the future, I was hoping to add support for Azure Kubernetes Service (AKS) and Google Kubernetes Engine (GKE).